src/troute_model.py

```diff
@@ -78,7 +91,20 @@ def preprocess_static_vars(self, values: dict):
             from_files=False, value_dict=values,
             segment_attributes=self._segment_attributes, 
             waterbody_attributes=self._waterbody_attributes)
+        
+        if len(values['upstream_id'])>0:
+            for key in values['upstream_id']:
+                del self._network._connections[key]
+                del self._network._reverse_network[key]
+                for tw in self._network._independent_networks.keys():
+                    del self._network._independent_networks[tw][key]
+                    for rli, _ in enumerate(self._network._reaches_by_tw[tw]):
+                        self._network._reaches_by_tw[tw][rli].remove(key)
+
+        network_end_time = time.time()                          # end of recording time for network creation #
+        self.task_times['network_time'] += network_end_time - network_start_time
 
+        DA_start_time = time.time()                             # start of recording time for data assimilation creation #
```

Comment:
We call time.time() twice here basically in a row. We could rewrite this to only call it once and use that same time call for both network_time and data_assimilation_time

Reply:
Great point. Basically DA_start_time = network_end_time
I'll delete line 107 and use network_end_time